libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -101,21 +101,13 @@ string IRGenerator::generate(ContractDefinition const& _contract)
 	t("memoryInit", memoryInit());
 	t("constructor", constructorCode(_contract));
 	t("deploy", deployCode(_contract));
-	// We generate code for all functions and rely on the optimizer to remove them again
-	// TODO it would probably be better to only generate functions when internalDispatch or
-	// virtualFunctionName is called - same below.
-	for (auto const* contract: _contract.annotation().linearizedBaseContracts)
-		for (auto const* fun: contract->definedFunctions())
-			generateFunction(*fun);
+	collectFunctions(_contract);
 	t("functions", m_context.functionCollector().requestedFunctions());
 
 	resetContext(_contract);
-	m_context.setMostDerivedContract(_contract);
```

Comment:
Don't we need this?

Reply:
it's called inside `resetContext` (he mentioned that earlier somewhere)

Reply:
We do. But `resetContext()` calls it too.